core/tests/performance_framework/perf_services.py

```diff
@@ -206,12 +206,33 @@ def get_page_timings_from_cached_session(self, page_url):
             page_session_stats=None, page_session_timings=page_session_timings)
 
     def _wait_until_page_load_is_finished(self, time_duration_secs=None):
+        """Waits for the complete page to load, otherwise XHR requests
+        made post initial page load will not be recorded.
+
+        Args:
+            time_duration_secs: int. Waiting time for page load. Defaults to
+            None.
+        """
         # Waits for the complete page to load, otherwise XHR requests
         # made post initial page load will not be recorded.
         time.sleep(time_duration_secs or self.DEFAULT_WAIT_DURATION_SECS)
 
     def _setup_proxy_server(self, downstream_kbps=None, upstream_kbps=None,
                             latency=None):
+        """Sets up a browsermobproxy server.
+
+        Args:
+            downstream_kbps: int. The downstream speed in kbps. Defaults to
+            None.
+            upstream_kbps: int. The upstream speed in kbps. Defaults to None.
+            latency: int. The latency of the server. Defaults to None.
+
+        Returns:
+            server: browsermobproxy Server instance. The browsermobproxy
```

Comment:
Hi @vibhor98 I am facing serious trouble in carrying out the tests for the type. I am currently using the simple approach of using cmd to call the python method from the class. I have downloaded the browsermob-proxy binary and have pointed the bat file when initializing the server in python shell. But it throws an error when I call `a._setup_proxy_server()` where `a=p.SeleniumPerformanceDataFetcher('firefox', 'NONE')` :
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "core/tests/performance_framework/perf_services.py", line 237, in _setup_proxy_server
    server.start()
  File "/home/jipkate/.local/lib/python2.7/site-packages/browsermobproxy/server.py", line 113, in start
    stderr=subprocess.STDOUT)
  File "/usr/lib/python2.7/subprocess.py", line 711, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1343, in _execute_child
    raise child_exception
OSError: [Errno 13] Permission denied
```
Please help me out here as I have tried Googling but in vain.
Thanks!

Reply:
@vibhor98 I am stuck here.

Reply:
Hi @YashJipkate, I just realized that these are all the test files where addition of args and returns is optional.  So, you can simply write `server` here in Returns section without adding instance. On a more general note, if you need to print type in some files, try to add `print type(var_name)` at relevant place in the method, start your local dev server (localhost) and trigger that action (sometimes may be navigating to some page, submitting an answer, running any job etc.) so that that method is called ultimately. Finally, look at the terminal for the type info (printed with error logs). 

Can you address all other comments as well as this PR is important and need to get into ASAP? Thanks!